doc/architecture/decisions/0010-use-repeatable-migrations-for-dev-and-test.md

```diff
@@ -0,0 +1,86 @@
+# 10. use_repeatable_migrations_for_dev_and_test
+
+Date: 2023-01-17
+
+## Status
+
+Accepted
+
+## Context
+
+We need to ensure that our development and test environments have
+predictable data, so we can log in, run tests and carry out usability
+testing sessions with users. As we run automated tests on dev, it's also
+important that the environment doesn't get clogged with automatically
+created data.
+
+## Decision
+
+We will use [repeatable Flyway migrations](https://flywaydb.org/blog/flyway-timestampsandrepeatables)
+to tear down and set up the data to a known state.
+
+These migrations will be in the format:
+
+```text
+R__$NUM_$NAME_OF_MIGRATION.sql
+```
+
+Where `$NUM` is the order we want the migration to be run in, and
+`$NAME_OF_MIGRATION` is a description of the migration to be run.
+
+Each migration will also have a comment at the top like so:
+
+```sql
+-- ${flyway:timestamp}
+```
+
+This changes the checksum for the repeatable migrations, meaning they
+get executed each time we migrate.
+
+In dev, we will have SQL migrations manually written to drop the
+tables and insert the data previously written in the previous data
+migrations. We will also drop the bookings table to remove any
+bookings created in the e2e tests.
+
+For test, we have a Wiremock repo, which mocks away external services.
+This repo also has a helper script, which uses mock data to generate /
+re-generate migrations based on any changes made to the mocks.
+
+When data changes in the mocks, we will run:
+
+```bash
+script/generate_migrations /path/to/api/directory
```

Comment:
Small typo on "generate_migration"

Reply:
Ah, sorry, this command is in the Wiremock repo. Have updated the copy to be clearer.